rasa/core/policies/ted_policy.py

```diff
@@ -228,276 +248,311 @@ def _load_params(self, **kwargs: Dict[Text, Any]) -> None:
     # data helpers
     # noinspection PyPep8Naming
     @staticmethod
-    def _label_ids_for_Y(data_Y: np.ndarray) -> np.ndarray:
-        """Prepare Y data for training: extract label_ids.
-        label_ids are indices of labels, while `data_Y` contains one-hot encodings.
+    def _surface_attributes(
+        features: List[List[Dict[Text, List["Features"]]]]
```

Comment:
there is many different types of `features` depending where `Dict` is, let's try to create names that reflect that

Reply:
do you have any suggestions?

Reply:
in single state featurizer, we use `features` for `"Features"`, all_features for `List["Features"]`, `featurized_state` for `Dict[Text, List["Features"]]`, I suggest using the same names for things before `_surface_attribute`, after it is surfaced maybe we can call it smth like `raw_data`, the rest seems to be fine

Reply:
In single state featurizer we have:
```
list_of_features                -> List["Features"]
state_features                 -> Dict[Text, List["Features"]
dialogue_state_features -> List[Dict[Text, List["Features"]]]
tracker_state_features   -> List[List[Dict[Text, List["Features"]]]]
```

after _` surface_attribute´:
```
state_to_tracker_features -> Dict[Text, List[List[List["Features"]]]]
tracker_features                -> List[List[List["Features"]]]
dialogue_features              -> List[List["Features"]]
turn_features                     -> List["Features"]
```
Does that work?

Reply:
usually we'd use `tracker_` and `dialogue_` as nearly synonyms. Does one more list correspond to examples?

Reply:
I guess we can also change `dialogue_features` to `turn_features` then.

Reply:
I think it is correct now